CHANGELOG.md

```diff
@@ -1,4 +1,35 @@
 # Release History
+## 0.7.0 (2022-05-25)
+
+### New Features:
+- Added support for user-defined table functions (UDTFs).
+  - Use function `snowflake.snowpark.functions.udtf()` to register a UDTF, or use it as a decorator to register the UDTF.
+  - Alternatively, use `Session.udtf.register()` to register a UDTF.
+  - Use `Session.udtf.register_from_file()` to register a UDTF from a Python file.
+
+### Improvements:
+- Updated APIs to query a table function, including both Snowflake built-in table functions and UDTFs.
+  - Use function `snowflake.snowpark.functions.table_function()` to create a callable representing a table function and use it to call the table function in a query.
+  - Alternatively, use function `snowflake.snowpark.functions.call_table_function()` to call a table function.
+  - Added support for `over` clause that specify `partition by` and `order by` when lateral joining a table function.
+  - Updated `Session.table_function()` and `DataFrame.join_table_function()` to accept `TableFunctionCall` instances.
+  - Deprecated `Session.flatten()` and `DataFrame.flatten()`.
+- Ensured that UDFs and UDTFs will be thread-safe by adding a lock to the function when it is called for the first time per thread.
```

Comment:
This statement isn't accurate. UDFs are not thread-safe after this change because the lock is only used when a UDF handler is called for the first time in a thread. It prevents problems (performance or other problems) when multiple UDF calls are started up by multiple threads.

Reply:
changed to `Added a lock to UDFs and UDTFs when it is called for the first time per thread`